File: magellan-legacy/src/main/java/com/wealthfront/magellan/EventTracker.java
package com.wealthfront.magellan;

class EventTracker {

  private EvictingQueue<Event> events;

  EventTracker(final int maxSize) {
    events = new EvictingQueue<>(maxSize);
  }

  void reportEvent(Event event) {
    events.add(event);
  }

  String getEventsDescription() {
    StringBuilder eventsDescription = new StringBuilder();
    for (Event event : events) {
      eventsDescription.append(event);
      eventsDescription.append("\n");
    }
    return eventsDescription.toString();
  }

}

File: magellan-kotlin/src/main/java/com/wealthfront/magellan/kotlin/EventTracker.kt
package com.wealthfront.magellan.kotlin

internal class EventTracker(maxSize: Int) {

  private val events: EvictingQueue<Event> = EvictingQueue(maxSize)

  val eventsDescription: String
    get() {
      val eventsDescription = StringBuilder()
      events.forEach {
        eventsDescription.append("$it \n")
      }
      return eventsDescription.toString()
    }

  fun reportEvent(event: Event) {
    events.add(event)
  }

}
